bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/process/monitors/StreamsProxy.java

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2020 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Ali Azam Rana - Added some extension required for IDF Plugin
+ *******************************************************************************/
+
+package com.espressif.idf.debug.gdbjtag.openocd.dsf.process.monitors;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+import org.eclipse.debug.core.model.IBinaryStreamMonitor;
+import org.eclipse.debug.core.model.IBinaryStreamsProxy;
+import org.eclipse.debug.core.model.IProcess;
+import org.eclipse.debug.core.model.IStreamMonitor;
+import org.eclipse.debug.internal.core.InputStreamMonitor;
+
+import com.espressif.idf.debug.gdbjtag.openocd.dsf.process.StreamListener;
+
+@SuppressWarnings("restriction")
+public class StreamsProxy implements IBinaryStreamsProxy {
```

Comment:
If this is a derived class, could you mention the original class details and what part of the code is modified in that for the future code maintenance point of view?

Reply:
done